core/trino-main/src/main/java/io/trino/operator/LookupJoinOperator.java

```diff
@@ -162,6 +164,7 @@ public void close()
         private final LookupJoinPageBuilder pageBuilder;
         private final Map<Integer, SavedRow> spilledRows = new HashMap<>();
         private final boolean probeOnOuterSide;
+        private final boolean matchSingleBuildRow;
```

Comment:
I'd call this "outputSingleMatch". It's not about matching a single row, but outputting a single row from all the ones that match.